qcodes/instrument_drivers/stahl/stahl.py

```diff
@@ -0,0 +1,220 @@
+            send_string,
+            datatype='B',
+            header_fmt="empty"
+        )
+
+        chnr = self._channel_number - 1
+        channel_group = chnr // 4
+        lock_code_group = response[channel_group]
+        return format(lock_code_group, "b")[chnr % 4 + 1] == "1"
+
+
+class Stahl(VisaInstrument):
+    """
+    Stahl driver.
+
+    Args:
+        name
+        address: A serial port address
+    """
+    def __init__(self, name: str, address: str):
+        super().__init__(name, address, terminator="\r")
+        self.visa_handle.baud_rate = 115200
+
+        instrument_info = self._parse_idn_string(
+            self.ask("IDN")
+        )
+
+        for key, value in instrument_info.items():
+            setattr(self, key, value)
+
+        channels = ChannelList(
+            self, "channel", StahlChannel, snapshotable=False
+        )
+
+        for channel_number in range(1, self.n_channels + 1):
+            name = f"channel{channel_number}"
+            channel = StahlChannel(
+                self,
+                name,
+                channel_number
+            )
+            self.add_submodule(name, channel)
+            channels.append(channel)
+
+        self.add_submodule("channel", channels)
+
+        self.add_parameter(
+            "temperature",
+            get_cmd=self._get_temperature,
+            unit="C"
+        )
+
+        self.connect_message()
+
+    def _get_temperature(self) -> float:
+        """
+        When querying the temperature a non-ascii character to denote
+        the degree symbol '°' is present in the return string. For this
+        reason, we cannot use the `ask` method as it will attempt to
+        decode the response with utf-8, which will fail. We need to
+        manually set the decoding to latin-1
+        """
+        send_string = f"{self.identifier} TEMP"
+        self.write(send_string)
+        response = self.visa_handle.read(encoding="latin-1")
+        temperature_string, = re.search("TEMP (.*)°C", response).groups()
+        return float(temperature_string)
+
+    @staticmethod
+    def _parse_idn_string(ind_string) -> Dict[str, Any]:
+        """
+        Return:
+             dict with keys: "model", "serial_number", "voltage_range",
+             "n_channels", "output_type"
+        """
+        groups = re.search(
+            "(HV|BS)(\d{3}) (\d{3}) (\d{2}) [buqsm]",
+            ind_string
+        ).groups()
+
+        id_parsers: Dict[str, Callable] = {
+            "model": str,
+            "serial_number": str,
+            "voltage_range": float,
+            "n_channels": int,
+            "output_type": {
+                "b": "bipolar",
+                "u": "unipolar",
+                "q": "quadrupole",
+                "s": "steerer",
+                "m": "bipolar milivolt"
+            }.get
+        }
+
+        return {
+            name: id_parsers[name](value)
+            for name, value in zip(id_parsers.keys(), groups)
```

Comment:
I think that, to be on the safe side, you should make the dict a OrderedDict when using zip with the keys and some other iterable.